### PR TITLE
dialects: Stencil dialect minor enhancement

### PR DESCRIPTION
**Pull request by mesham**
All in experimental
LB and UB on stencil apply set and removal of external store in lowering

### CODE REVIEW

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py [hunk 1]
```diff
@@ -303,6 +304,14 @@ def match_and_rewrite(self, op: ExternalLoadOp, rewriter: PatternRewriter,
         pass
 
 
+class TrivialExternalStoreOpCleanup(RewritePattern):
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ExternalStoreOp, rewriter: PatternRewriter,
+                          /):
+            rewriter.replace_matched_op([], [])
```
Comment (webmiche) [inline xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:312]: Why not erase the op? :sweat_smile:

Comment (mesham) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:312]: what's the API call for this?

Comment (PapyChacal) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py:312]: `rewriter.erase_matched_op()` :slightly_smiling_face:

**Review by PapyChacal — APPROVED** [review]
Looks good!

**Review by superlopuh — APPROVED** [review]

File: xdsl/transforms/experimental/ConvertStencilToLLMLIR.py (hunk 1 above)
Comment (mesham) [reply xdsl/transforms/experimental/ConvertStencilToLLMLIR.py]: Thanks, done :)

**Pull request merged**